contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1316,7 +1316,10 @@ get_physical_user_name(char *db_name, char *user_name, bool suppress_error)
 			if ((strlen(user_name) == 3 && strncmp(user_name, "dbo", 3) == 0) ||
 				(strlen(user_name) == 8 && strncmp(user_name, "db_owner", 8) == 0))
 			{
-				return new_user_name;
+				if(suppress_role_error || user_exists_for_db(db_name, new_user_name))
```

Comment:
Can't we merge all this in a single if condition?

Reply:
Done